### ISSUE TITLE
`expectCall` cheatcode overloaded with `msg.value` assertion

### ISSUE THREAD

**Issue by tynes** [foundry-rs/foundry#1568]
### Component

Forge

### Describe the feature you would like

Currently `vm.expectCall` operates on a target address and the calldata it expects

```solidity
 function expectCall(address,bytes calldata) external;
```

When sending ETH through multiple contracts, it would be useful to have an overloaded `expectCall` that also accepts a `value` param that also asserts on call context's value (`msg.value`)

```solidity
function expectCall(address,uint256 value,bytes calldata) external;
```

Didn't see this mentioned in any open issues, including https://github.com/foundry-rs/foundry/issues/876

**Comment by AdithyaNarayan** [comment]
Is this issue available? I would like to work on this as my first PR

**Comment by tynes** [comment]
I think overloading `mockCall` in the same way would be useful too

**Comment by AdithyaNarayan** [comment]
While overloading `mockCall`, I came across bug #1604 . Is it ok if I do one PR to fix both of them? (Since #1604 is a one line fix)

**Comment by AdithyaNarayan** [comment]
Quick question, for the `mockCall` overload, which would take precedence, `msg.value`, or calldata?

```solidity
vm.mockCall(address(target), 10, abi.encodeWithSelector(target.f.selector), abi.encode(10));
vm.mockCall(address(target), abi.encodeWithSelector(target.f.selector, 1), abi.encode(20));

assertEq(target.f{value: 10}(1) == ???);
``` 

What is the expected return data in this case?
